### PR TITLE
GEOMESA-844 Set recommended table configuration options on create

### PR DESCRIPTION
**Pull request by anthonyccri**
* Set table.split.threshold to 128M
* Set table.block.cache.enabled to true
* NOTE: maxShards has been somewhat deprecated

Signed-off-by: Anthony Fox <anthonyfox@ccri.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
looks good.

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/tables/SpatioTemporalTable.scala [hunk 1]
```diff
@@ -86,4 +90,20 @@ object SpatioTemporalTable extends GeoMesaTable with Logging {
     bd.delete()
     bd.close()
   }
+
+
+  override def configureTable(sft: SimpleFeatureType, tableName: String, tableOps: TableOperations): Unit = {
+    // NOTE: since the ST table is likely going away, I'm not inclined to thread maxShards all the way through
+    // the call chain so I just set a default of 40 here
```
Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/tables/SpatioTemporalTable.scala]: I'm fine with this right now.  In order to codify routing extras info, how about widening configureTable to take a Map[String, String]?

Comment (anthonyccri) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/tables/SpatioTemporalTable.scala]: Yeah, I thought about that and possibly just using some hints in the user data of the SFT.  Let's push that off till later though.

Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/tables/SpatioTemporalTable.scala]: Ah, user data, of course.  And yes, future us have lots of to do.

**Comment by anthonyccri** [comment]
Merged.

**Pull request closed without merging**